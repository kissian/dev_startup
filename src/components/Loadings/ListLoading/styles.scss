@mixin nth-children($points...) {
  @each $point in $points {
      &:nth-child(#{$point}) {
          @content;
      }
  }
}

.textInputLoading {
  height: 214px;
  border: 1px solid #eee;
  border-radius: 0;
  width: 100%;

  .loadingLine {
      height: 10px;
      margin: 10px;
      animation: pulse 1s infinite ease-in-out;
  }

  div {
      @include nth-children(1, 5, 9) {
          width: 40%;
      }
      @include nth-children(2, 6, 10) {
          width: 80%;
      }
      @include nth-children(3, 7) {
          width: 50%;
      }
      @include nth-children(4, 8) {
          width: 60%;
      }
  }
}

@keyframes pulse {
  0% {
      background-color: rgba(165, 165, 165, 0.1)
  }
  50% {
      background-color: rgba(165, 165, 165, 0.3)
  }
  100% {
      background-color: rgba(165, 165, 165, 0.1)
  }
}