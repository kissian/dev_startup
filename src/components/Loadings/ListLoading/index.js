import React from 'react';
import './styles.scss';

const ListLoading = () => (

    <div className="textInputLoading">
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
        <div className="loadingLine" />
    </div>

);

export default ListLoading;
