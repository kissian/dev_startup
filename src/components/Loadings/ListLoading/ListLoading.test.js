import React from 'react';
import { shallow } from 'enzyme';
import ListLoading from './index';

describe('ListLoading component', () => {
    let component = null;
    it('renders component', () => {
        component = shallow(<ListLoading />);
    });
    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
