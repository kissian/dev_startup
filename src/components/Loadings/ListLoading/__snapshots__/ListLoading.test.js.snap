// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLoading component matches snapshot 1`] = `
<div
  className="textInputLoading"
>
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
  <div
    className="loadingLine"
  />
</div>
`;
